emperor/support_files/js/shapes.js

```diff
@@ -3,18 +3,53 @@
 // shapes.
 define(['jquery', 'three'], function($, THREE) {
 
-  var shapes = {
-    'sphere': new THREE.SphereGeometry(0.1, 8, 8),
-    'cube': new THREE.CubeGeometry(0.2, 0.2, 0.2, 8, 8, 8),
-    'cone': new THREE.CylinderGeometry(0.1, 0, 0.1, 10),
-    'icosahedron': new THREE.IcosahedronGeometry(0.1, 0),
-    'cylinder': new THREE.CylinderGeometry(0.05, 0.05, 0.2, 10)
-  };
+  var SPHERE = 'Sphere', CUBE = 'Cube', CONE = 'Cone',
+      ICOSAHEDRON = 'Icosahedron', CYLINDER = 'Cylinder';
+
+  var shapes = [SPHERE, CUBE, CONE, ICOSAHEDRON, CYLINDER];
+
+  /*
+   *
+   * Return a correctly sized geometry that matches the plotting space
+   *
+   * @param {shapeName} string, one of 'Sphere', 'Cube', 'Cone', 'Icosahedron'
+   * or 'Cylinder'.
+   * @param {ranges} object, with two arrays of "max" and "min" values as
+   * generated by the DecompositionModel's dimensionRanges property;
+   *
+   * @return THREE.Geometry, returns the requested geometry object with a size
+   * appropriate for the data presented on screen.
+   *
+   **/
+  function getGeometry(shapeName, ranges){
+
+    // this is a heauristic tested on numerous plots since 2013, based off of
+    // the old implementation of emperor. We select the dimensions of all the
+    // geometries based on this factor.
+    var factor = (ranges.max[0] - ranges.min[0]) * 0.012;
+
+    switch(shapeName){
+      case SPHERE:
+        return new THREE.SphereGeometry(factor, 8, 8);
+      case CUBE:
+        return new THREE.CubeGeometry(factor, factor, factor, 8, 8, 8);
+      case CONE:
+        return new THREE.CylinderGeometry(factor * 0.4, 0, 1.5 * factor, 8);
+      case ICOSAHEDRON:
+        return new THREE.IcosahedronGeometry(factor, 0);
+      case CYLINDER:
+        return new THREE.CylinderGeometry(factor * 0.4, factor * 0.4,
+                                          1.5 * factor, 10);
+      default:
+        throw Error("Unknown geometry requested: " + shapeName);
+    }
+  }
 
   var $shapesDropdown = $('<select>');
-  for (shape in shapes) {
+  _.each(shapes, function(shape){
     $shapesDropdown.append(new Option(shape, shape));
-  }
+  });
 
-  return {shapes: shapes, $shapesDropdown: $shapesDropdown};
+  return {$shapesDropdown: $shapesDropdown,
```

Comment:
We should also still return the shapes list so we have a list of what is available.